jitbuilder/compile/Method.hpp

```diff
@@ -166,7 +167,7 @@ class ResolvedMethod : public ResolvedMethodBase, public Method
                                TR::SymbolReferenceTable *symRefTab);
 
    protected:
-   const char *_fileName;
+   std::string _fileName;
```

Comment:
Because the JIT is not supposed to allocate memory on the malloc heap. Compiler classes do not have destructors to manage memory the way you're imagining. You need to allocate these objects using an Allocator so that any supplementary memory allocated comes from the JIT allocator rather than, say, ::new() . 

Take a look at the definition and the constructor's initialization of, say, the field `_dependentSymRefs` in `compiler/optimizer/RematTools.hpp`